raster/r.proj/testsuite/test_r_proj.py

```diff
@@ -0,0 +1,302 @@
+                "null_cells": 0,
+                "cells": 324000,
+                "min": 55.9677429199219,
+                "max": 156.223129272461,
+                "range": 100.255386352539,
+                "mean": 110.375287905093,
+                "mean_of_abs": 110.375287905093,
+                "stddev": 20.3150368008109,
+                "variance": 412.700720218302,
+                "coeff_var": 18.4054213460164,
+                "sum": 35761593.28125,
+            },
+            "lanczos_f": {
+                "n": 324000,
+                "null_cells": 0,
+                "cells": 324000,
+                "min": 55.9677429199219,
+                "max": 156.221099853516,
+                "range": 100.253356933594,
+                "mean": 110.375363786639,
+                "mean_of_abs": 110.375363786639,
+                "stddev": 20.3150742610671,
+                "variance": 412.702242232673,
+                "coeff_var": 18.4054426315072,
+                "sum": 35761617.8668709,
+            },
+        },
+    }
+
+    @classmethod
+    def setUpClass(cls):
+        cls.use_temp_region()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.del_temp_region()
+        if cls.to_clear:
+            cls.runModule(
+                "g.remove", flags="f", type="raster", name=",".join(cls.to_clear)
+            )
+
+    def test_same_resolution(self):
+        self.runModule("g.region", raster="elevation")
+        for method, reference in self.references["same"].items():
+            output_file = "elevation_same_{}".format(method)
+            self.to_clear.append(output_file)
+            self.runModule(
+                "r.proj",
+                input="elevation",
+                location=self.location,
+                output=output_file,
+                method=method,
+            )
+            self.assertRasterFitsInfo(
+                raster=output_file,
+                reference={
+                    "north": 228500,
+                    "south": 215000,
+                    "east": 645000,
+                    "west": 630000,
+                    "nsres": 10,
+                    "ewres": 10,
+                    "rows": 1350,
+                    "cols": 1500,
+                    "cells": 2025000,
+                    "datatype": "FCELL",
+                    "ncats": 0,
+                },
+            )
+            self.assertRasterFitsUnivar(
+                raster=output_file,
+                reference=reference,
+                precision=1e-5,
+            )
+
+    def test_different_resolution(self):
+        self.runModule("g.region", raster="elevation", res=25)
+        for method, reference in self.references["different"].items():
+            output_file = "elevation_diff_{}".format(method)
+            self.to_clear.append(output_file)
+            self.runModule(
+                "r.proj",
+                input="elevation",
+                location=self.location,
```

Comment:
I think you will need to add `mapset="PERMANENT"`.

Reply:
Noted!